Foundation/HTTPCookieStorage.swift

```diff
@@ -62,22 +62,27 @@ open class HTTPCookieStorage: NSObject {
     }
     private let syncQ = DispatchQueue(label: "org.swift.HTTPCookieStorage.syncQ")
 
-    private init(cookieStorageName: String) {
+    private let isEphemeral: Bool
+
+    private init(cookieStorageName: String, isEphemeral: Bool = false) {
         _allCookies = [:]
         cookieAcceptPolicy = .always
+        self.isEphemeral = isEphemeral
         super.init()
-        let bundlePath = Bundle.main.bundlePath
-        var bundleName = bundlePath.components(separatedBy: "/").last!
-        if let range = bundleName.range(of: ".", options: .backwards, range: nil, locale: nil) {
-            bundleName = String(bundleName[..<range.lowerBound])
+        if !isEphemeral {
+            let bundlePath = Bundle.main.bundlePath
+            var bundleName = bundlePath.components(separatedBy: "/").last!
+            if let range = bundleName.range(of: ".", options: .backwards, range: nil, locale: nil) {
+                bundleName = String(bundleName[..<range.lowerBound])
+            }
+            let cookieFolderPath = _CFXDGCreateDataHomePath()._swiftObject + "/" + bundleName
```

Comment:
I am still in the learning phase, ignore this comment if it is wrong,

After looking the code from `TestHTTPCookieStorage.test_cookieInXDGSpecPath()`, it seems you are missing,

```swift
if let xdg_data_home = getenv("XDG_DATA_HOME") {
   // XDG Path
} else {
   // Non XDG Path
}
```

Reply:
@karthikkeyan For ephemeral storage, we don't persist the cookies in a file. The XDG spec standardises some common locations on the file system. It's not relevant if we don't want to store cookies to a file. I hope I got your question right.

Reply:
Sorry for the late response, the statement that creates the cookie folder's path, is inside `if !isEphemeral`
```swift
if !isEphemeral {
    // ...

    let cookieFolderPath = _CFXDGCreateDataHomePath()._swiftObject + "/" + bundleName

    // ...
}
```

We can leave that discussion for different PR as the code is not relevant to the intention of this PR.